core/src/main/java/org/elasticsearch/index/translog/Checkpoint.java

```diff
@@ -33,25 +34,40 @@
 
 /**
  */
-class Checkpoint {
+public class Checkpoint {
+
+    static final byte FORMAT_VERSION = 1;
 
    static final int BUFFER_SIZE = Integer.BYTES  // ops
             + Long.BYTES // offset
-            + Long.BYTES;// generation
+            + Long.BYTES // generation
+            + Long.BYTES; // seq no global checkpoint
     final long offset;
     final int numOps;
     final long generation;
+    final long seqNoGlobalCheckpoint;
 
-    Checkpoint(long offset, int numOps, long generation) {
+    Checkpoint(long offset, int numOps, long generation, long seqNoGlobalCheckpoint) {
```

Comment:
@s1monw and I discussed this and came up with this alternative plan:

1. we have a Translog.setGlobalSeqNoCheckpoint() which updates a field in the current writer
2. we add a special op which contains the current checkpoint and has a fixed known length
3. we write the op before we fsync
4. the offset of the translog after the above write goes in the the checkpoint file
5. we have a static “readGlobalSeqNoCheckpoint” which opens the .ckp file and the corresponding tlog and reads the above struct and returns the value